src/lib/services/events.js

```diff
@@ -0,0 +1,99 @@
+// const moment = require('moment');
+const repo = require('../connectors/repos');
+const { mapValues, mapKeys } = require('lodash');
+const camelCase = require('camelcase');
+const snakeCase = require('snake-case');
+const Event = require('../models/event');
+
+/**
+ * Checks whether the supplied value/key is a JSON field and is not null
+ * @param {Mixed} value
+ * @param {String} key
+ * @return {Boolean} true if key is a JSON field and value is not null
+ */
+const isMappedField = (value, key) => {
+  const jsonFields = ['licences', 'entities', 'metadata'];
+  return jsonFields.includes(key) && (value !== null);
+};
+
+/**
+ * Maps keys of the data using the supplied mapper function
+ * @param  {Object} data     - data to map
+ * @param  {Function} mapper - function to use for mapping keys
+ * @return {Object}        data with keys mapped
+ */
+const keyMapper = (data, mapper) => {
+  return mapKeys(data, (value, key) => mapper(key));
+};
+
+/**
+ * Maps data to the format expected by the repository class
+ * @param  {Object} data - event data
+ * @return {Object}      - event data mapped to DB column names/types
+ */
+const mapToRepo = data => {
+  const mapped = mapValues(data, (value, key) => {
+    return isMappedField(value, key) ? JSON.stringify(value) : value;
```

Comment:
There is a Bookshelf plugin for defining which fields are JSON:
https://github.com/seegno/bookshelf-json-columns
So JSON.stringify could perhaps be handled at the repository layer using this plugin

Reply:
done